packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +98,162 @@ describe("Pooled Transaction Fees Test", function () {
 			RUNTIME_ERRORS.TRANSACTION.FEES_NOT_ENOUGH
 		);
 	});
+
+	it("Should pass if two out of three doesn't pay the txn fee in a group txn of size three", () => {
+		const amount = 4000;
+		const fee = 3000;
+		// group with fee distribution. Pooled transaction fee
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: bob.account,
+				toAccountAddr: john.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(alice.balance(), BigInt(initialBalance));
+		assert.equal(bob.balance(), BigInt(initialBalance));
+	});
+
+	it("Should unfunded accounts be able to issue transactions and opt-in", () => {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elon.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			},
+			{
+				type: types.TransactionType.OptInASA,
+				sign: types.SignType.SecretKey,
+				fromAccount: elon.account, // unfunded account
+				assetID: assetId,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert(elon.balance() !== BigInt(0));
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(amount));
+		assert.equal(alice.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(elon.balance(), BigInt(amount)); // unfunded account
+		// verify holding
+		assert.isDefined(elon.getAssetHolding(assetId));
+	});
+
+	it("Should fail when no fund is sent to unfunded account in group txn", () => {
```

Comment:
I think it should not fail. Same as the one below. This is a new feature that we do not support in algob yet.

Reply:
We can update the test later once it is supported by algob. The current test is based on what is supported yet. It is expected to fail as of now.

Reply:
I think the goal of this task is to support it

Reply:
We really need to confirm, if this is really what is meant by an unfunded account. In my opinion, it's related to pooling txn.

Reply:
I found an article: https://developer.algorand.org/articles/avm-7-new-features/?from_query=pooling

What I understood from here is that: The unfunded accounts can call Smart Contracts and interact with a dapp, while not being funded, as long as they are grouped with a fee paying transaction.

Reply:
I have added support for this.